src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -487,7 +487,7 @@ private void EmitBinaryExpressionNode(AssociativeNode node)
                 EmitOpCode(OpCodes.Ldarg_2);
                 EmitOpCode(OpCodes.Ldstr, lNode.Value);
                 // if t is a single value, wrap it in an array of the single value.
-                if (!typeof(IEnumerable).IsAssignableFrom(t))
+                if (!typeof(IEnumerable).IsAssignableFrom(t) || typeof(string).IsAssignableFrom(t))
```

Comment:
So is this when you encounter an AST initializing a string array - an `ExprListNode `of strings?

Reply:
Or is it when emitting the individual strings in that expression list?

Reply:
no, this is when we encounter a single string - `system.string` implements `IEnumerable` so it previously did not get wrapped in an outer `IList` - weird things then happen, because the output Dictionary is of type `<String,IList>` - at runtime the CLR in some cases will show that the value is an `IList` and in others it will show that it's a list, very strange things happen when trying to access the underlying object - I guess because it's a reference that it can work until you try to resolve the reference, I would have though it would fail when the CLR jited it.